### PR TITLE
Add need for PDMC in cli-device-management.md

### PR DESCRIPTION
**Pull request by AnotherButler**
Add need for PDMC and bootloader, as requested in issue #707

### CODE REVIEW

File: docs/tools/offline/cli-device-management.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 Arm Mbed OS allows you to update your device firmware, enabled by the Pelion IoT platform. Mbed CLI includes features to prepare and ship updates for devices managed through the [Device Management Portal](https://cloud.mbed.com/docs/current/introduction/index.html). Mbed CLI provides the subcommand `mbed device-management` to manage devices (`mbed dev-mgmt` and `mbed dm` are also available as shorter aliases). The remainder of this document uses the `mbed dm` alias for all device management subcommands. This document explains the steps to enable and use Pelion Device Management with a project.
```
Comment (peknis01) [inline docs/tools/offline/cli-device-management.md]: Pelion Device Management IoT platform, when mentioned for the first time. Then just Device Management (no Pelion).

**Comment by AnotherButler** [comment]
Thanks @peknis01 Please re-review.

File: docs/tools/offline/cli-device-management.md [hunk 2]
```diff
@@ -87,3 +93,5 @@ Once you execute `mbed dm update prepare`, Mbed CLI automatically uploads the pa
 ### Advanced use
 
 Mbed CLI allows for significantly more flexibility than the model above shows in exactly the same way as [the manifest tool](https://cloud.mbed.com/docs/current/updating-firmware/manifest-tool.html). You can override each of the defaults that `mbed dm init` sets by using the command-line or an input file. Mbed CLI supports a variety of commands. You can print a full list of commands by using `manifest-tool --help`.
+
+To test this feature, you can run the [Device Management Client Example](https://github.com/ARMmbed/mbed-cloud-client-example) on one of our reference platforms.
```
Comment (teetak01) [inline docs/tools/offline/cli-device-management.md]: version 2.0.0 or later. (2.0.0 is not released yet).

I would also link to the Cloud-side tutorial.

Comment (teetak01) [reply docs/tools/offline/cli-device-management.md]: Why is this mentioned in Advanced usage? This is the only possible way to use the dm feature.

Already in requirements you should link to Cloud-side documentation + quick-starts when those are ready.

**Comment by AnotherButler** [comment]
@teetak01 Thanks for the review. Do my changes resolve the remaining issues you found?

**Pull request merged**